crates/wasmtime/src/store.rs

```diff
@@ -230,6 +231,18 @@ enum CallHookInner<T> {
     Async(Box<dyn CallHookHandler<T> + Send + Sync>),
 }
 
+/// What to do after returning from a deadline callback.
+pub enum DeadlineResult {
```

Comment:
Maybe

```suggestion
pub enum DeadlineBehavior {
```

to make `Result<DeadlineBehavior>`?

Another option is to have this return a `std::ops::ControlFlow` where the break is an `anyhow::Error` and the continue is either a sync or async enum with a `u64` payload. This would also allow for `?` usage. But my experience with `ControlFlow` has not been super great.